README.md

```diff
@@ -1,12 +1,108 @@
 # Opencensus
 
+[![CircleCI](https://circleci.com/gh/opencensus-beam/opencensus_elixir.svg?style=svg)](https://circleci.com/gh/opencensus-beam/opencensus_elixir)
+[![Hex version badge](https://img.shields.io/hexpm/v/opencensus_elixir.svg)](https://hex.pm/packages/opencensus_elixir)
+
+Wraps some [`:opencensus`][:opencensus] capabilities for Elixir users so
+they don't have to [learn them some Erlang][LYSE] in order to get
+[OpenCensus] distributed tracing.
+
+[OpenCensus]: http://opencensus.io
+[:opencensus]: https://hex.pm/packages/opencensus
+[LYSE]: https://learnyousomeerlang.com
+
 ## Installation
 
+Add `opencensus_elixir` to your `deps` in `mix.exs`:
+
 ```elixir
 def deps do
   [
-    {:opencensus_elixir, "~> 0.2.0"}
+    {:opencensus, "~> 0.9"},
+    {:opencensus_elixir, "~> 0.3.0"}
   ]
 end
 ```
 
+## Usage
+
+Wrap your code with the
+[`Opencensus.Trace.with_child_span/3`][oce-with_child_span-3] macro to
+execute it in a fresh span:
+
+```elixir
+import Opencensus.Trace
+
+def traced_fn(arg) do
+  with_child_span "traced" do
+    :YOUR_CODE_HERE
+  end
+end
+```
+
+That's the rough equivalent of the following direct use of `:opencensus`:
+
+```elixir
+def traced_fn() do
+  try do
+    :ocp.with_child_span("traced", %{fn: :traced_fn, mod: __MODULE__})
+    {:span_ctx, trace_id, span_id, trace_options, _} = :ocp.current_span_ctx()
+
+    Logger.metadata(
+      trace_id: :io_lib.format("~32.16.0b", [trace_id]) |> List.to_string(),
+      span_id: :io_lib.format("~16.16.0b", [span_id]) |> List.to_string(),
+      trace_options: trace_options
+    )
+
+    :YOUR_CODE_HERE
+  after
+    :ocp.finish_span()
+    Logger.metadata(trace_id: nil, span_id: nil, trace_options: nil)
+  end
+end
+```
```

Comment:
I do not think that that expansion is needed there. Especially as it can provide wrong view on what is happening. It should be hidden from the user.

Reply:
Guess so; depends on how well we maintain it. Even if we don’t document this as what the macro does, I reckon we should document it as what they’ll have to do if they _don’t_ use the macro. Shoving it into a guide on how to drive `:ocp` properly from Elixir might suffice.

Zooming out a little, developers are going to search for “OpenCensus Elixir” and expect what they find to make it clear how to do one with the other. We need more than just our `deps` and a few one-line `@doc` strings.

Reply:
this of course can go out of sync. Maybe pseudo code outline will be better?

Reply:
I think linking to segments of the documentation of opencensus_erlang, even though it’s not the *best* experience for some elixir devs, would be the only maintainable option going forward.

Reply:
> this of course can go out of sync

I trust we’ll pick up on drift in our own code during review: we’ve got, what, literally everyone weighing in here just on changes to the README? :) 

I’ve raised #16 to help protect us against `:opencensus` drift: pseudo-code and doc-links could become just as wrong. I’ll document the version for which our code/docs work, and see what else I can tighten up.

How stable is the `:ocp` API, in particular? The merger might well _extend_ it… reckon they’ll take a breaking change?

Reply:
I just don’t see why anyone who comes to this library would really need to know what it would look like if you used the underlying erlang library. If they want to know that, they can just go to that libraries documentation, right?

Reply:
I can see us linking to the relevant locations in that documentation of course.

Reply:
Looping back — has it been a _month_!? — I'm leaning more to @hauleth' stance. I've cut down the lengthy guided tour in the README to a short description of how to drive `:opencensus` from Elixir should one be so inclined.